src/SchemaFactory.php

```diff
@@ -391,7 +391,7 @@ public function createSchema(): Schema
         $inputTypeUtils     = new InputTypeUtils($annotationReader, $namingStrategy);
         $inputTypeGenerator = new InputTypeGenerator($inputTypeUtils, $fieldsBuilder);
 
-        if (empty($this->typeNamespaces) && empty($this->typeMappers)) {
+        if (empty($this->typeNamespaces) && empty($this->typeMappers) && empty($this->typeMapperFactories)) {
```

Comment:
Why are we adding an additional condition here for throwing an exception?  This is relaxing the state where an exception is thrown, requiring that the `typeMapperFactories` also evaluate to false.

Reply:
I wanna use SchemaFactory only with `addTypeMapperFactory`, without calling addTypeMapper.

Reply:
I don't think this is the right way to go about this.  Is a factory a requirement for the `SchemaFactory` to work properly?  Why can't i just pass in `typeNamespaces` and `typeMappers` only?  Keep in mind that your commits apply to everyone using this lib and not everyone is going to use it the same.

I'm still not quite sure what you're trying to accomplish here by throwing an exception if the factory isn't present.

Reply:
If you intend on only relying on the factories, maybe the better logic is to check to see if there are factories present, process those to get the type mappers, assigning them to the `typeMappers` property, and then leaving the exception as it is, checking to see if `typeMappers` and `typeNamespaces` are both not empty.

Reply:
IMHO this is valid example, how i want to use this lib. But its not possible now.

```php
    $mapperFactory = new StaticClassListTypeMapperFactory($classList);
    $queryProviderFactory = new AggregateControllerQueryProviderFactory($controllers, $container);
    $factory = new SchemaFactory($cache, $container);
    $factory->addQueryProviderFactory($queryProviderFactory);
    $factory->addTypeMapperFactory($mapperFactory);
    $schema = $factory->createSchema();
```

Reply:
@janatjak I think that's a valid expectation.  Please see my previous comment.  The logic here needs to be written to be flexible and cover all scenarios, not just this one.  In your code, if I do not include a factory, I'll get an exception and it shouldn't be a requirement to include a factory, right?

Reply:
I'm sorry, I missed your prev comment. Pls check my last commit. Thanks

Reply:
Why not process the factories first and add the type mappers from the factory to the class property `typeMappers` instead of creating an additional variable?  The logic looks good though, just could be a little cleaner I think.

Reply:
Doing what I'm suggesting would mean you'd only need to add the `$compositeTypeMapper->addTypeMapper($typeMapper);` call within the loop on `typeMappers`.

Reply:
good point, thank you for your hint